File: spring_java_orm_sample/src/main/java/com/example/demo/jooq/dsl/PublisherDsl.java
package com.example.demo.jooq.dsl;

import lombok.RequiredArgsConstructor;
import org.jooq.DSLContext;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;

import static com.example.demo.jooq.generated.Tables.PUBLISHER;

@Component
@Profile("jooq")
@RequiredArgsConstructor
public class PublisherDsl {
    private final DSLContext dsl;

    public boolean existsPublisher(Long publisherId) {
        return dsl.fetchExists(PUBLISHER, PUBLISHER.ID.eq(publisherId));
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/jooq/dsl/PublisherDsl.kt
package com.example.demo.jooq.dsl

import com.example.demo.jooq.generated.Tables
import org.jooq.DSLContext
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component

@Component
@Profile("jooq")
class PublisherDsl(private val dsl: DSLContext) {
    fun existsPublisher(publisherId: Long?): Boolean {
        return dsl.fetchExists(Tables.PUBLISHER, Tables.PUBLISHER.ID.eq(publisherId))
    }
}
